app/controllers/api/generic_works_controller.rb

```diff
@@ -0,0 +1,92 @@
+module API
+  # Adds an endpoint that customizes the update action on nested works
+  class GenericWorksController < ApplicationController
+    before_action :my_load_and_authorize_resource, only: [:update, :destroy, :show, :add_parent, :remove_parent]
```

Comment:
So the only action that isn't in there is #create. Should we change it to 
```
before_action :my_load_and_authorize_resource, except: [:create]
```
as a better alternative?